src/js/modules/RoutingApp.js

```diff
@@ -25,186 +28,174 @@ import {
 } from "./core/components";
 import { DescriptionPage } from "./descriptions/components";
 import { SectionPage } from "./sections/components";
-import {
-  ContributorPage,
-  PhotographerPage,
-  IllustratorPage,
-} from "./users/components";
+import { ContributorPage, ArtistPage } from "./users/components";
+
+import { sessionify } from "./accounts/actions";
 
-import { fetchAllData } from "./core/actions";
-import { sessionfy } from "./accounts/actions";
+const RoutingAppQuery = gql`
+  query RoutingAppQuery {
+    allSections {
+      id
+      name
+      permalink
+    }
+  }
+`;
 
-class RoutingApp extends Component {
+class RoutingApp extends PureComponent {
   constructor(props) {
     super(props);
   }
 
   componentDidMount() {
-    this.prepareData();
-  }
+    const { sessionify } = this.props;
 
-  // Separate function because it is reused for the DataErrorPage
-  prepareData = () => {
-    this.props.fetchAllData();
     const session = localStorage.getItem("session");
     if (session) {
-      this.props.sessionfy(JSON.parse(session));
+      sessionify(JSON.parse(session));
+    } else {
+      const urlHeaders = queryString.parse(window.location.search);
+      sessionHeaders = ["client_id", "token", "uid"];
+      if (sessionHeaders.every(header => header in urlHeaders)) {
+        sessionify({
+          "access-token": urlHeaders.token,
+          client: urlHeaders.client_id,
+          uid: urlHeaders.uid,
+        });
+      }
     }
-  };
+  }
 
   render() {
-    const {
-      sections,
-      descriptions,
-      session,
-      isAllDataFetched,
-      initDataError,
-    } = this.props;
+    const { data: { loading, error, allSections }, session } = this.props;
+
+    if (document.getElementById("loading") && !loading) {

```

Comment:
Why not create a variable outside the if condition? Technically you can also do
`if ((const loadingIcon = document.getElementById("loading")) && !loading)` but I find that less readable

Reply:
Also if you do this in render, every time RoutingApp rerenders, you're running this code. Which...I'm not so sure about. Why not do it in the component lifecycle?

Reply:
Data might not come until a few tenths of a second after the component mounts and renders.